internal/storage/storage.go

```diff
@@ -25,3 +44,43 @@ type Blobstore interface {
 	// DeleteObject deltes an object or does nothing if the object doesn't exist.
 	DeleteObject(ctx context.Context, bucket, objectName string) error
 }
+
+// Creates a new Blobstore using the provided context
+type BlobstoreFactory func(ctx context.Context) (Blobstore, error)
+
+// Blobstore that does nothing.
+type NoopBlobstore struct{}
+
+func NewNoopBlobstore(ctx context.Context) (Blobstore, error) {
+	return &NoopBlobstore{}, nil
+}
+
+// No op.
+func (s *NoopBlobstore) CreateObject(ctx context.Context, folder, filename string, contents []byte) error {
+	return nil
+}
+
+// No op.
+func (s *NoopBlobstore) DeleteObject(ctx context.Context, folder, filename string) error {
+	return nil
+}
+
+// Creates a new BlobstoreFactory based on the provided BlobstoreType.
+func NewBlobstoreFactory(blobstoreType BlobstoreType) BlobstoreFactory {
```

Comment:
yeah - if you change the config provider func to return a string and not a factory - this can just be a method that is something like

func CreateBlobstore - accepting the blobstore type